gusto/initialisation_tools.py

```diff
@@ -275,15 +276,28 @@ def calculate_Pi0(state, theta0, rho0):
     return Pi0
 
 
-def moist_hydrostatic_balance(state, theta_e, water_t, pi_boundary=Constant(1.0)):
+def saturated_hydrostatic_balance(state, theta_e, water_t, pi0=None,
+                                  top=False, pi_boundary=Constant(1.0),
+                                  max_iterations=40):
     """
     Given a wet equivalent potential temperature, theta_e, and the total moisture
     content, water_t, compute a hydrostatically balance virtual potential temperature,
     dry density and water vapour profile.
+
+    The general strategy is to split up the solving into two steps:
+    1) finding rho to balance the theta profile
+    2) finding theta_v and r_v to get back theta_e and saturation
+    We iteratively solve these steps until we (hopefully)
+    converge to a solution.
+
     :arg state: The :class:`State` object.
     :arg theta_e: The initial wet equivalent potential temperature profile.
     :arg water_t: The total water pseudo-mixing ratio profile.
-    :arg pi_boundary: the value of pi on the lower boundary of the domain.
+    :arg pi0: Optional function to put exner pressure into.
+    :arg top: If True, set a boundary condition at the top, otherwise
+              it will be at the bottom.
+    :arg pi_boundary: The value of pi on the specified boundary.
+    :arg max_iterations: Max number of iterations for balance solver.
```

Comment:
I don't think this is an accurate docstring for the `max_iterations` arg. Some may confuse this with iterations of the Krylov method to invert the linear system. Something like:

```python
"""
...
:arg max_outer_solve_count: Maximum number of outer linear solves for the entire balance solver.
:arg max_inner_solve_count: Maximum number of inner linear solves for the entire balance solver.
"""
```
```

Reply:
Done, thanks!